feature/interface/singleton/ate_tests/singleton_test/singleton_test.go

```diff
@@ -417,17 +430,24 @@ func (tc *testCase) testFlow(t *testing.T, packetSize uint16, ipHeader ondatra.H
 	} else if avg := octets / outPkts; avg > uint64(tc.mtu) {
 		t.Errorf("Flow source packet size average got %d, want <= %d (MTU)", avg, tc.mtu)
 	}
-	if p1InDiff.unicast < outPkts {
-		t.Errorf("DUT received too few source packets: got %d, want >= %d", p1InDiff.unicast, outPkts)
-	}
 
 	if inPkts == 0 {
 		// The PacketLargerThanMTU cases do not expect to receive packets,
-		// so this is not an error.
+		// so this is not an error. Ethernet counters will be incremented
+		// in case of Packets which larger than MTU
 		t.Log("Flow did not receive any packet")
-	} else if avg := octets / inPkts; avg > uint64(tc.mtu) {
-		t.Errorf("Flow destination packet size average got %d, want <= %d (MTU)", avg, tc.mtu)
+		if p1InEthDiff.unicast < outPkts {
+			t.Errorf("DUT received too few source packets: got %d, want >= %d", p1InEthDiff.unicast, outPkts)
```

Comment:
this message has no difference with line 444. Can you please make them different and more precise to differentiate the error cases?

Reply:
updated please check.